src/sentry/search/eap/spans/aggregates.py

```diff
@@ -27,16 +27,54 @@ def resolve_count_op_filter(op_value: str) -> TraceItemFilter:
             value=AttributeValue(val_str=op_value),
         )
     )
+    return (AttributeKey(name="sentry.op", type=AttributeKey.TYPE_STRING), filter)
+
+
+def resolve_key_eq_value_filter(
+    aggregate_key: AttributeKey, key: AttributeKey, value: str
+) -> tuple[AttributeKey, TraceItemFilter]:
+    filter = TraceItemFilter(
+        comparison_filter=ComparisonFilter(
+            key=key,
+            op=ComparisonFilter.OP_EQUALS,
+            value=AttributeValue(val_str=value),
+        )
+    )
+    return (aggregate_key, filter)
 
 
 SPAN_CONDITIONAL_AGGREGATE_DEFINITIONS = {
     "count_op": ConditionalAggregateDefinition(
         internal_function=Function.FUNCTION_COUNT,
         default_search_type="integer",
         arguments=[ArgumentDefinition(argument_types={"string"}, is_attribute=False)],
-        key=AttributeKey(type=AttributeKey.TYPE_STRING, name="sentry.op"),
-        filter_resolver=resolve_count_op_filter,
-    )
+        aggregate_resolver=resolve_count_op,
+    ),
+    "avg_if": ConditionalAggregateDefinition(
+        internal_function=Function.FUNCTION_AVG,
+        default_search_type="duration",
+        arguments=[
+            ArgumentDefinition(
+                argument_types={
+                    "duration",
+                    "number",
+                    "percentage",
+                    *constants.SIZE_TYPE,
+                    *constants.DURATION_TYPE,
+                },
+            ),
+            ArgumentDefinition(
+                argument_types={
+                    "string",
+                },
+            ),
```

Comment:
Isn't this arg specifically columns only and not just a freeform string

Reply:
Yeah, `ArgumentDefinition` automatically has `is_attribute=True`, which means it should be a column. "string" in this case means that the value of that column should be a "string".
 This means that the second parameter can't be something like 'span.duration'. This is needed because the `resolve_key_eq_value_filter` assumes `value` is a string.
 
 Now that you point this out, I do see how this could get confusing. 
1. When `ArgumentDefinition.is_attribute = False`, `argument_types` is the type of the argument itself.
 2. When `ArgumentDefinition.is_attribute=True`, `argument_types` is the type stored in that column. You can already assume the argument itself is a string because it should just be a column name.
 
 Perhaps after this PR, we can discuss how to make this less confusing?

Reply:
I also added a test cases to reiterate that avg_if shouldn't accept a numeric column in the second argument